### PR TITLE
Issue #9: As a user, I want to filter my shopping list to make it easier to locate an item in the list. 🔍📜

### LINKED ISSUE #9: 9. As a user, I want to filter my shopping list to make it easier to locate an item in the list.
AC:

* Display a text field above the top of the shopping list
* As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
* When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
* The filter text should match any part of the item name (i.e. it should _not_ only match from the start of the string)

### PR DESCRIPTION
**Pull request by thekidnamedkd**
## Description

This PR focuses on implementing a 'search bar' form to filter a user's current list of items. It adds a filter function using state to evaluate the list of itemName's against the current user form input and returns only matching strings in the render. From there we also added an 'X' button to clear the form by resetting the form value/filterValue to an empty string. We chose this approach because it fits nicely into our existing component and it creates a 'real time search' behavior on the local list returned from the database query. 

## Related Issue

closes #9 

## Acceptance Criteria

* Display a text field above the top of the shopping list
* As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
* When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
* The filter text should match any part of the item name (i.e. it should _not_ only match from the start of the string)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Screen Shot 2021-01-27 at 5 36 23 PM

### After

filter list
## Testing Steps / QA Criteria

* pull the latest commit of this branch into your local IDE
* run the code on a local server in the browser
* create a new list or work with an existing list if you have a token in local storage
* use different combinations of key inputs in the search to find items in your list
* click the X button to clear the form and return to a full list view

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,7 @@
 /build
 
 # misc
+.vscode
```
Comment (jessicasalbert) [inline .gitignore:15]: What does this do? Just curious!

File: src/components/ShoppingList/PopulatedList.js [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
+
+      setFilteredList(filtered);
+    }
+  }, [filterValue, listData]);
+
+  const onFilterChange = (e) => {
+    setFilterValue(e.target.value);
+  };
+
+  const resetFilter = () => {
```
Comment (jessicasalbert) [inline src/components/ShoppingList/PopulatedList.js:33]: Love this lil feature!

File: src/components/ShoppingList/PopulatedList.js [hunk 3]
```diff
@@ -1,15 +1,39 @@
...
+  const [filterValue, setFilterValue] = useState("");
+  const [filteredList, setFilteredList] = useState([]);
+
+  useEffect(() => {
+    if (listData) {
+      let filtered = listData.docs.filter((doc) => {
+        return doc
+          .data()
+          .itemName.toLowerCase()
+          .includes(filterValue.toLowerCase());
```
Comment (jessicasalbert) [inline src/components/ShoppingList/PopulatedList.js:22]: This is super nice! So with the two dependency arrays, the filtered list will be updated any time A) there's a change to the DB data or B) the filter search term changes. Which is perfect - great job.

File: src/components/ShoppingList/PopulatedList.js [hunk 4]
```diff
@@ -1,15 +1,39 @@
-import React, { Fragment } from "react";
+import React, { Fragment, useState, useEffect } from "react";
 import { useCollection } from "react-firebase-hooks/firestore";
 import { db } from "../../lib/firebase";
 
 const PopulatedList = () => {
-  const [value, loading, error] = useCollection(
+  const [listData] = useCollection(
```
Comment (jessicasalbert) [inline src/components/ShoppingList/PopulatedList.js:6]: I like how you cleaned this part up (and the conditional rendering down below)

**Review by jessicasalbert — APPROVED** [review]
Awesome work! I'm still learning useEffect but I was able to follow your code easily and it helped me understand a bit more too. Super clean and works as expected per my tests. Nice work controlling for capitalization too.

File: .gitignore (hunk 1 above)
Comment (thekidnamedkd) [reply .gitignore:15]: Just makes sure VS code local settings don't make it into the development repo.

File: src/components/ShoppingList/PopulatedList.js (hunk 3 above)
Comment (wlcreate) [inline src/components/ShoppingList/PopulatedList.js:22]: Love that you're adding the extra check for ```if``` there is a list and then comparing each list item to the search term by lowercasing both to make sure they're the same!

**Review by wlcreate — APPROVED** [review]
Great job! I like how you both cleaned up PopulatedList's ```useCollection``` hook and are keeping track of the changes to the search term to affect what from the list is being rendered. Also, I just realized that we're including ```aria-label``` for the input and button for accessibility 😍

**Review by MelinaMejia95 — APPROVED** [review]

**Pull request merged**